File: app/src/main/java/com/example/babytimes/CreateAccount.java
package com.example.babytimes;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.DialogInterface;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.ArrayList;

public class CreateAccount extends AppCompatActivity {

    RecyclerView recyclerView;
    FloatingActionButton add_button;
    ImageView empty_imageview;
    TextView no_data;

    DatabaseHelper myDB;
    ArrayList<String> user_id, user_firstName, user_lastName, user_email, user_username, user_password;
    CustomAdapter customAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.createaccount);

        recyclerView = findViewById(R.id.recyclerView);
        add_button = findViewById(R.id.add_button);
        empty_imageview = findViewById(R.id.empty_imageview);
        no_data = findViewById(R.id.no_data);
        add_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(CreateAccount.this, AddUserActivity.class);
                startActivity(intent);
            }
        });

        myDB = new DatabaseHelper(CreateAccount.this);
        user_id = new ArrayList<>();
        user_firstName = new ArrayList<>();
        user_lastName = new ArrayList<>();
        user_email = new ArrayList<>();
        user_username = new ArrayList<>();
        user_password = new ArrayList<>();

        storeDataInArrays();

        customAdapter = new CustomAdapter(CreateAccount.this,this, user_id, user_firstName, user_lastName,
                user_email, user_username, user_password);
        recyclerView.setAdapter(customAdapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(CreateAccount.this));

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if(requestCode == 1){
            recreate();
        }
    }

    void storeDataInArrays(){
        Cursor cursor = myDB.readAllData();
        if(cursor.getCount() == 0){
            empty_imageview.setVisibility(View.VISIBLE);
            no_data.setVisibility(View.VISIBLE);
        }else{
            while (cursor.moveToNext()){
                user_id.add(cursor.getString(0));
                user_firstName.add(cursor.getString(1));
                user_lastName.add(cursor.getString(2));
                user_email.add(cursor.getString(3));
                user_username.add(cursor.getString(4));
                user_password.add(cursor.getString(5));
            }
            empty_imageview.setVisibility(View.GONE);
            no_data.setVisibility(View.GONE);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.my_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == R.id.delete_all){
            confirmDialog();
        }
        return super.onOptionsItemSelected(item);
    }

    void confirmDialog(){
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle("Delete All?");
        builder.setMessage("Are you sure you want to delete all Data?");
        builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                DatabaseHelper myDB = new DatabaseHelper(CreateAccount.this);
                myDB.deleteAllData();
                //Refresh Activity
                Intent intent = new Intent(CreateAccount.this, CreateAccount.class);
                startActivity(intent);
                finish();
            }
        });
        builder.setNegativeButton("No", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {

            }
        });
        builder.create().show();
    }
}

File: app/src/main/java/com/example/babytimes/CreateAccount.kt
package com.example.babytimes;

import android.content.Intent;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import java.sql.SQLException // for database SQL functions

import android.os.Bundle;
import android.provider.ContactsContract
import android.service.autofill.UserData
import android.util.Log
import android.widget.TextView;
import androidx.core.content.ContextCompat.startActivity;
import java.sql.Timestamp
import java.time.LocalDateTime

class CreateAccount  : AppCompatActivity() {
    private val dbHelper = DatabaseHelper()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.createaccount)

    }

    fun MainMenu (view: View) {
        val intent = Intent(this, MainMenu::class.java)
        startActivity(intent)
    }

    //PUSH NEW USER
    fun NewUser() {
        Log.d("CreateAccount", "NewUser function called")
        val currentDateTime = LocalDateTime.now()
        val timestamp = Timestamp.valueOf(currentDateTime.toString()) // Convert LocalDateTime to a Timestamp
        insertUser("John", "Doe", timestamp, 0, "p@ssword", "jdoe@asu.edu")
    }
    // Create an instance of the DatabaseHelper class


    fun insertUser(firstName: String, lastName: String, createdDate: java.sql.Timestamp, isDeleted: Int, Password: String, Email: String) {
        val sql = "USE BabyTimes;" +
                "INSERT INTO users (FirstName, LastName, CreatedDate, IsDeleted, Password, Email) VALUES (?, ?, ?, ?, ?, ?)"

        try {
            val preparedStatement = dbHelper.connection.prepareStatement(sql)

            preparedStatement.setString(1, firstName)
            preparedStatement.setString(2, lastName)
            preparedStatement.setTimestamp(3, createdDate)
            preparedStatement.setInt(4, isDeleted)
            preparedStatement.setString(5, Password)
            preparedStatement.setString(6, Email)

            val rowsInserted = preparedStatement.executeUpdate()

            if (rowsInserted > 0) {
                println("User inserted successfully.")
            } else {
                println("Insertion failed.")
            }

            preparedStatement.close()
        } catch (e: SQLException) {
            e.printStackTrace()
        } finally {
            dbHelper.closeConnection()
        }
        //MainMenu()
    }

}

